generation/earl-grey.js

```diff
@@ -0,0 +1,167 @@
+const t = require("babel-types");
+const objectiveCParser = require("objective-c-parser");
+const generate = require("babel-generator").default;
+const fs = require("fs");
+const {
+  isNumber,
+  isString,
+  isBoolean,
+  isPoint,
+  isOneOf,
+} = require('./type-checks');
+
+function createClass(json) {
+  return t.classDeclaration(
+    t.identifier(json.name),
+    null,
+    t.classBody(json.methods.map(createMethod.bind(null, json.name))),
+    []
+  );
+}
+
+function createExport(json) {
+  return t.expressionStatement(
+    t.assignmentExpression(
+      "=",
+      t.memberExpression(
+        t.identifier("module"),
+        t.identifier("exports"),
+        false
+      ),
+      t.identifier(json.name)
+    )
+  );
+}
+
+function capitalizeFirstLetter(string) {
```

Comment:
it might be a good time to move `capitalizeFirstLetter` and `methodNameToSnakeCase` and `sanitizeArgument` either to another file that specifically is for helper methods, or at least organise them together so that the main functions in this file (`createMethod`, `createClass`, ...) are logically sorted for future reading of this file.